### PR TITLE
chore:remove specia code for react 16

### PR DESCRIPTION
**Pull request by likezero**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/3330)

### CODE REVIEW

File: components/input-item/PropsType.tsx [hunk 1]
```diff
@@ -35,4 +35,5 @@ export interface InputItemPropsType {
   onFocus?: InputEventHandler;
   onBlur?: InputEventHandler;
   onVirtualKeyboardConfirm?: InputEventHandler;
+  disabledKeys?: Array<any> | null,
```
Comment (ziluo) [inline components/input-item/PropsType.tsx]: 这里强校验一下吧，类似下面
```ts
type InputKey = '0'|'1'|'2'|'3'|'4'|'5'|'6'|'7'|'8'|'9'|'.'
disabledKeys?: Array<InputKey> | null 
```

Comment (likezero) [reply components/input-item/PropsType.tsx]: 不用呀。includes方法会判断。
你这样写的话。InputKey 在渲染键盘的时候有一个配置，这里又加一个重复的。

**Comment by ziluo** [comment]
+1

File: components/input-item/CustomInput.tsx [hunk 2]
```diff
@@ -120,50 +109,37 @@ class NumberInput extends React.Component<NumberInputProps, any> {
...
+      document.body.appendChild(container);
     }
+    this.container = container;
     return this.container;
   }
 
   renderCustomKeyboard() {
-    if (IS_REACT_16) {
-      return;
-    }
```
Comment (ziluo) [inline components/input-item/CustomInput.tsx]: 这里不是要切换成 React.createPortal 么？

Comment (likezero) [reply components/input-item/CustomInput.tsx]: 先不改了

File: components/input-item/CustomInput.tsx [hunk 3]
```diff
@@ -86,7 +87,6 @@ class NumberInput extends React.Component<NumberInputProps, any> {
   }
 
   componentDidUpdate() {
```
Comment (ziluo) [inline components/input-item/CustomInput.tsx]: 这里的空函数删了吧

**Comment by ziluo** [comment]
+1

**Pull request merged**